lib/crypto/tasks.py

```diff
@@ -1,18 +1,102 @@
+        # and installs it.
+        bump_version_number(version)
         try:
             version.sign_files()
         except SigningError as e:
-            log.warning('Failed signing version %s: %s.' % (version.pk, e))
+            log.warning(
+                'Failed signing version {0}: {1}.'.format(version.pk, e))
+
+
+def bump_version_number(version):
+    """Add a .1 to the version number in the install.rdf or package.json."""
+    for file_obj in version.files.all():  # Bump the numbers in each file.
+        # Create a new xpi with the bumped version.
+        bumped = '{0}.bumped'.format(file_obj.file_path)
+        # Copy the original XPI, with the updated install.rdf or package.json.
+        with ZipFile(file_obj.file_path, 'r') as source:
+            file_list = source.infolist()
+            with ZipFile(bumped, 'w', zipfile.ZIP_DEFLATED) as dest:
+                for file_ in file_list:
+                    content = source.read(file_.filename)
+                    if file_.filename == 'install.rdf':
+                        content = _bump_version_in_install_rdf(content)
+                    if file_.filename == 'package.json':
+                        content = _bump_version_in_package_json(content)
+                    dest.writestr(file_, content)
+        # Move the bumped file to the original file.
+        shutil.move(bumped, file_obj.file_path)
+    # Now update the Version model.
+    version.update(version='{0}.1'.format(version.version))
+
+
+def _dot_one(version):
+    """Returns the version with an appended .1 on it."""
+    return '{0}.1'.format(version)
+
+
+def _bump_version_in_install_rdf(content):
+    """Add a '.1' to the version number in the install.rdf provided."""
+    # We need to use an XML parser, and not a RDF parser, because our
+    # install.rdf files aren't really standard (they use default namespaces,
+    # don't namespace the "about" attribute... rdflib can parse them, and can
+    # now even serialize them, but the end result could be very different from
+    # the format we need.
+    tree = etree.fromstring(content)
+    # There's two different formats for the install.rdf: the "standard" one
+    # uses nodes for each item (like <em:version>1.2</em:version>), the other
+    # alternate one sets attributes on the <RDF:Description
+    # RDF:about="urn:mozilla:install-manifest"> element.
+
+    # Get the version node, if it's the common format, or the Description node
+    # that has the "em:version" attribute if it's the alternate format.
+    node = tree.xpath(
+        '//em:version | //*[@em:version]',
+        namespaces={'em': 'http://www.mozilla.org/2004/em-rdf#'})[0]
+    if node.tag.endswith('version'):  # Common format, version is a node.
+        node.text = _dot_one(node.text)
+    else:  # Alternate format, version is an attribute.
+        version_uri = '{http://www.mozilla.org/2004/em-rdf#}version'
```

Comment:
This API makes me sad...

Anyway, can we move this definition outside of the `if` and use `node.tag == version_uri` rather than `endswith`?